### ISSUE TITLE
5.4.4-beta.2 and future releases

### ISSUE THREAD

**Issue by kytwb** [getferdi/ferdi#416]
Before publishing:

- [x] Changelog
- [x] Binary testing

Let me know if the current tag is acceptable or if we want to include more things (in which case we can still delete the existing draft release, retag a new commit and let the CI recreate a new draft release with the updated assets). We have to be careful not to publish releases that don't have all the assets file uploaded from both AppVeyor/Travis CI as it happened in the past:

<img width="363" alt="Screen Shot 2020-02-29 at 1 23 20 AM" src="https://user-images.githubusercontent.com/412895/75596720-17d85500-5a92-11ea-895e-b885c7519bcb.png">

The draft is expected to have 13 attached assets (at the time of writing). When you publish the draft, GitHub will add 2 more assets, bringing the total to 15 (source code zip/tar.gz).  
The idea of having release publish peer-reviewing a-la pull request was [submitted to GitHub](https://twitter.com/kytwb/status/1222627736918855685).

(1) Regarding the changelog: for now we do it manually. I suggest we start exploring automatic changelog generation based on a commits message template. That would involve making most of the contributions (exceptions aside) on adequate branch, pull requests and peer review (like we started to do more recently), and proper squash merge to develop with the changelog in mind when producing the commit message that will land on develop.

(2) I don't think this is necessary, but to be precautious, I recommend testing at least the produced binary for your platform before the publisher of the release hit the publish button, to double-check we don't get segfaults at startup like it happened before because of the app signing/notarization on macOS. We probably can do without, but lacking a test suite for now, we can't be too precautious.

Any feedback/opinion/additional comment welcome!
cc @vantezzen

**Comment by kytwb** [comment]
We may want to address https://github.com/getferdi/ferdi/issues/329 in this release or as a high priority for the next one as it seems the auto updater has been broken since a dependency upgrade.

**Comment by kytwb** [comment]
We can already start making use of [staged rollouts](https://www.electron.build/auto-update#staged-rollouts) also if needed.

**Comment by kytwb** [comment]
Current status:

<img width="903" alt="Screen Shot 2020-03-01 at 11 40 41 PM" src="https://user-images.githubusercontent.com/412895/75635602-b04c1200-5c17-11ea-9870-247957792d23.png">

**Comment by kytwb** [comment]
Let's look at making use of the announcements API for the next non-beta release.

**Comment by kytwb** [comment]
Released v5.4.4-beta.2 (with `stagingPercentage: 42` in the `latest*.yml` files). 
 I will increase to 100% in the coming days. 

We would need to establish which commit message style to use to generate the changelog automatically based on the pull requests merge commits. I propose the repository maintainers could/should

**Comment by kytwb** [comment]
Closing as [v5.4.4-beta.2] is published, but keeping the thread open to discussion awaiting @vantezzen (currently traveling) feedback. Others are also welcome to participate. We will open subsequent tickets to address things mention in this ticket (setup automatic changelog, commit style validation, maybe keep announcements in versioning as I saw they are [ignored](https://github.com/getferdi/server/blob/a58ffc02be614f5396f69e5c48ead872d059d6b1/.gitignore#L21) (?) etc.).

**Comment by vantezzen** [comment]
I think having a routine to follow before publishing is a good idea. Your suggested routine sounds good, I'd suggest we use:
- [ ] Open a new issue regarding the release of a new version (to make sure there are no features that have to be completed before releasing the version)
- [ ] Update Changelog (hopefully automatic, as suggested)
- [ ] Binary testing (hopefully also on a Windows VM?)
- [ ] Add announcement to the server for large full releases (Not really needed for beta versions)
- [ ] Stage rollout (50% -> 100%), I'd also suggest only updating the download link on the website once we confirm there are no large bugs in the release

I'd also suggest that we use the commit message style as suggested by [Chris Beams](https://chris.beams.io/posts/git-commit/#seven-rules).

I'm not very familiar with testing ElectronJS apps but we can also think about adding an automatic test suite to Ferdi that we can run before publishing to ensure basic functionality works.

WDYT?
